numpy/ma/core.py

```diff
@@ -3105,6 +3105,14 @@ def __getitem__(self, indx):
         Return the item described by i, as a masked array.
 
         """
+        # 2016.01.15 -- v1.11.0
+        warnings.warn(
+            "Currently, slicing will try to return a view of the data," +
+            " but will return a copy of the mask. In the future, it will try" +
+            " to return both as views.",
+            FutureWarning
+        )
+
```

Comment:
Minor comment: Do we need warnings in `__getitem__`? The proposed change shouldn't have any effect on someone who never modifies the mask, right? It's only in `__setitem__` that code might break.

Also, perhaps we can be even fancier and only show the warning if we know `self.data` is a view: In `__setitem__ do `if not self.data.owndata: warnings.warn('...')`.